include/openssl/arm_arch.h

```diff
@@ -78,6 +78,56 @@
 // ARMV8_SHA512 indicates support for hardware SHA-512 instructions.
 #define ARMV8_SHA512 (1 << 6)
 
+// ARMV8_SHA3 indicates support for hardware SHA-3 instructions including EOR3.
+#define ARMV8_SHA3  (1 << 11)
+
+// ARMV8_WIDE_AES_PMULL_PIPELINE indicates that the microarchitecture
+// supports a high unrolling factor of AES-GCM due to a wide AES/PMULL pipeline.
+#define ARMV8_WIDE_AES_PMULL_PIPELINE (1 << 12)
```

Comment:
Consider naming of this. Does it actually have anything to do with a specific AES instruction?

If not, can we give it a general name so we can use it to define function level distinguishers? I.e. I can use this for s2n-bignum integration but it looks a bit weird using something saying `aes`.